core/src/edu/asu/c3simulator/simulation/Simulation.java

```diff
@@ -0,0 +1,214 @@
+package edu.asu.c3simulator.simulation;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.PriorityQueue;
+
+import edu.asu.c3simulator.util.Observable;
+import edu.asu.c3simulator.util.ObservationListener;
+import edu.asu.c3simulator.util.Subroutine;
+
+/**
+ * Simulation that tracks time using an adjustable {@link #timeScale} and
+ * {@link #scaledTimeUnit}. Each millisecond of real-time represents {@link #timeScale}
+ * amount of time in-game, and the in-game time is expressed in {@link #scaledTimeUnit}s
+ * <p>
+ * {@link #update(float)} should be called regularly, as it is what will update the
+ * in-game time, and trigger all applicable simulation events.
+ * <p>
+ * DateEvents will be triggered whenever the in-game date is updated, and can be
+ * registered with {@link #registerDateListener(ObservationListener)}.
+ * <p>
+ * StateChangeEvents will be triggered whenever any change in the simulation data occurs,
+ * including date changes, changes in business value, new businesses starting, etc.
+ * <p>
+ * Some cases will require events to be triggered at a certain in-game time. For these
+ * cases, use {@link #registerTimedEvent(Date, Subroutine)}
+ * 
+ * @author Moore, Zachary
+ * @version 0.1
+ *
+ */
+public class Simulation implements C3Simulation, Observable<Simulation>
+{
+	private static class EventNode implements Comparable<EventNode>
+	{
+		private Date date;
+		private Subroutine event;
+		
+		@Override
+		public int compareTo(EventNode other)
+		{
+			return date.compareTo(other.date);
+		}
+	}
+	
+	private static final DateFormat DEFAULT_DATE_FORMAT = new SimpleDateFormat(
+			"dd MMM yy");
```

Comment:
Should that be 2 M's instead of 3 for the date? If that is not the case, why so? I'm assuming d = day, m = month, and y = year.